### ISSUE TITLE
TOC is broken when generating a libdoc html from XML:HTML format

### ISSUE THREAD

**Issue by Snooz82** [robotframework/robotframework#3708]
in RF 3.2.2 the Table of content is broken when generating a libdoc.html from a former generated xml that contains html docs.

```
python -m robot.libdoc SeleniumLibrary sl.libspec
python -m robot.libdoc sl.libspec sl.html
```

This is due to xmlwriter.py does use different html conversion than the htmlwriter and does not create links for html.

**Comment by Snooz82** [comment]
can be closed

Solved with #3716

**Comment by pekkaklarck** [comment]
The TOC here refers to the special `%TOC%` marker Libdoc supports since RF 3.2 (#3280) to generate the table of contents automatically.
